scripts/prepare_data.py

```diff
@@ -62,67 +164,120 @@ def add_tile_id(row):
     # Prepare the tiles
 
     ## Convert datasets shapefiles into geojson format
-    logger.info('Convert label shapefile into GeoJSON format (EPSG:4326)...')
-    labels = gpd.read_file(SHPFILE)
-    labels_4326 = labels.to_crs(epsg=4326)
-
-    nb_labels = len(labels)
-    logger.info('There is/are ' + str(nb_labels) + ' polygon(s) in ' + SHPFILE)
+    logger.info('Convert labels shapefile into GeoJSON format (EPSG:4326)...')
+    labels_gdf = gpd.read_file(SHPFILE)
+    if 'year' in labels_gdf.keys():
+        labels_gdf['year'] = labels_gdf.year.astype(int)
+        labels_4326_gdf = labels_gdf.to_crs(epsg=4326).drop_duplicates(subset=['geometry', 'year'])
+    else:
+        labels_4326_gdf = labels_gdf.to_crs(epsg=4326).drop_duplicates(subset=['geometry'])
+    nb_labels = len(labels_gdf)
```

Comment:
I would name the variable `nb_gt_labels`, because you use it later in the script where the label dataset have GT + FP. It's better to avoid confusion.